### PR TITLE
chore(Field.PhoneNumber): add tel input mode to phone number input

### PR DESCRIPTION
**Pull request by joakbjerk**
solves https://github.com/dnbexperience/eufemia/issues/3091

### CODE REVIEW

File: packages/dnb-eufemia/src/extensions/forms/Field/PhoneNumber/__tests__/PhoneNumber.test.tsx [hunk 1]
```diff
@@ -773,4 +773,20 @@ describe('Field.PhoneNumber', () => {
...
+      '.dnb-forms-field-phone-number__country-code .dnb-input__input'
+    )
+    const phoneNumberInput = document.querySelector(
+      '.dnb-forms-field-phone-number__number .dnb-input__input'
+    )
+
+    expect(countryCodeInput).not.toHaveAttribute('inputmode')
+
+    expect(phoneNumberInput).toHaveAttribute('inputmode')
+    expect(phoneNumberInput.getAttribute('inputmode')).toBe('tel')
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/extensions/forms/Field/PhoneNumber/__tests__/PhoneNumber.test.tsx:790]:
```suggestion
    expect(phoneNumberInput)).toHaveAttribute('inputmode', 'tel')
```

**Review by tujoworker — APPROVED** [review]

File: packages/dnb-eufemia/src/extensions/forms/Field/PhoneNumber/__tests__/PhoneNumber.test.tsx [hunk 2]
```diff
@@ -773,4 +773,20 @@ describe('Field.PhoneNumber', () => {
...
+    render(<PhoneNumber />)
+
+    const countryCodeInput = document.querySelector(
+      '.dnb-forms-field-phone-number__country-code .dnb-input__input'
+    )
+    const phoneNumberInput = document.querySelector(
+      '.dnb-forms-field-phone-number__number .dnb-input__input'
+    )
+
+    expect(countryCodeInput).not.toHaveAttribute('inputmode')
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/extensions/forms/Field/PhoneNumber/__tests__/PhoneNumber.test.tsx:787]: Not sure why we not should add inputmode of tel here ass well.

**Review by tujoworker — APPROVED** [review]

File: packages/dnb-eufemia/src/extensions/forms/Field/PhoneNumber/__tests__/PhoneNumber.test.tsx (hunk 2 above)
Comment (joakbjerk) [reply packages/dnb-eufemia/src/extensions/forms/Field/PhoneNumber/__tests__/PhoneNumber.test.tsx:787]: On android at least, I don't get a keyboard with letters, so I can't search by country name.

Comment (tujoworker) [reply packages/dnb-eufemia/src/extensions/forms/Field/PhoneNumber/__tests__/PhoneNumber.test.tsx]: You are right!

**Pull request merged**